CONTRIBUTING.md

```diff
@@ -0,0 +1,207 @@
+- Constants:```ALL_CAPS```
+
+#### Example
+```
+module circle_utils
+  implicit none
+  real, parameter :: PI = 3.14
+contains
+  function calculate_area(radius) result(area)
+    real, intent(in) :: radius
+    real :: area
+    area = PI * radius * radius
+  end function calculate_area
+end module circle_utils
+```
+
+### Java
+
+#### File Naming
+- Each class should go in its own ```.java``` file named after the class.
+
+#### Style
+- Follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+
+#### Naming
+- Classes: ```PascalCase```
+- Methods and variables: ```camelCase```
+- Constants: ```UPPER_SNAKE_CASE``` (with ```static final```)
+
+#### Example
+```
+public class Circle {
+    public static final double PI = 3.14;
+
+    public double calculateArea(double radius) {
+        return PI * radius * radius;
+    }
+}
+```
+### Python
+
+#### File Naming
+- Use ```snake_case.py``` for file names.
+
+#### Style
+- Follow guidelines in the [PEP 8](https://peps.python.org/pep-0008/).
+- Use 4-space indentation.
+- Limit lines to 79 characters.
+
+#### Naming
+- Variables and functions: ```snake_case```
+- Classes: ```PascalCase```
+- Constants: ```UPPER_SNAKE_CASE```
+
+#### Example
+```
+def calculate_area(radius):
+    """Calculate area of a circle."""
+    pi = 3.14
+    return pi*radius**2
+```
+
+## Branching Strategies
+
+### ```main``` Branch
+- Keep ```main``` always **stable and production-ready**.
+- Support **collaborative development**.
+- Ensure features and fixes are tested before release.
+- Require PR reviews.
+- Disallow direct pushes.
+
+## Git Operations and Pull Requests Guidelines
+
+### Branching
+
+- All branches should have a known goal and lifecycle.
+- The branch name should summarize its goal and be appended by:
+    - devops/ - branches with changes that do not actually affect source code
+    - feature/ - branches with new features
+    - bugfix/ - branches with bug fixes
+    - docs/ - for branches only containing documentation addition or edits
+- Branches should be lightweight according to the goal and lifecycle.
+- **Main branch must remain deployable at all times.**
+    - No pull request should be maid to main with partially completed work (unless coordinated beforehand).
+    - Deployable means:
+        - Developer tested
+        - Automated tests are passing
+        - Feature complete
+        - Peer-reviewed and accepted
+        - Static analysis completed
+        - Test coverage metrics goal met
+- **Other people's branches:**
+  1. Do not push commits to someone else's feature branch; at least not without coordinating with them first.
+     1. It is somewhat rude.
+     2. There could be local changes that the originator has not pushed up yet that you are missing.
+  2. Instead submit a PR to that branch if you feel you have a constructive change for it.
+- **Branches in general**
+    - Do not leave feature branches laying around.
+- **Avoid Git Force Pushes**
```

Comment:
This only make sense during code review (non-draft PR) as it messes with the ability to find "changes since last review"